pyccl/ccl_utils.i

```diff
@@ -49,4 +56,80 @@ void spline_integrate(int n_integ,
   free(_ys_in);
 }
 
+void get_spline1d_array_size(gsl_spline *spline, int* size, int* status) {
+  if(spline == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  if(spline->interp == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  *size = spline->interp->size;
+}
+
+void get_spline2d_array_sizes(gsl_spline2d *spline2d, int* x_size, int* y_size,
+                              int* status) {
+  if(spline2d == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  *x_size = spline2d->interp_object.xsize;
+  *y_size = spline2d->interp_object.ysize;
+}
+
+void get_spline1d_arrays(gsl_spline *spline,
+                         int x_size, double* xarr,
+                         int y_size, double* yarr,
+                         int *status)
+{
+  if(spline == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  if(x_size != spline->interp->size) {
+    *status = CCL_ERROR_INCONSISTENT;
+    return;
+  }
+  if(y_size != spline->interp->size) {
+    *status = CCL_ERROR_INCONSISTENT;
+    return;
+  }
+  for(int i=0; i<x_size; i++) {
```

Comment:
would a `memcpy` be faster here?

Reply:
same below for y

Reply:
Switched to `memcpy`, makes the code shorter as well. 
I played a bit around with godbolt (https://godbolt.org/z/13jKM6GE9) and for some compilers, `memcpy` and the loop end up with the same number of instructions. Interestingly, the loop gets more and more instructions with more recent compiler versions.
Since this gets copied to the python layer, I don't think any speed differences matter, if there are any.

Reply:
oh, that's interesting. Yes, I don't think this would ever be a bottleneck. Code looks neater though. Thanks!